### PR TITLE
Add actions for querying Authorize.net.

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Add actions for querying Authorize.net.
These actions are intended to be generic & usable on other processors. The main ones are

>$gateway = Omnipay::create('AIMGateway');
>$gateway->query(array('startTimestamp' => strtotime('yesterday'), 'endTimeStamp' => 'strtotime('now')));

or for subscription plans:
>$gateway->paymentPlansQuery();

The others are more internal.

The response object for query() supports
>$response->getFirstName();
>$response->getLastName();
>$response->getTransactionId();
// this is a standard transactionReference string, not the object
>$response->getTransactionReference();
>$response->getBillingAddress1();
>$response->getBillingAddress2();
>$response->getBillingPostcode();
>$response->getBillingState();
>$response->getBillingCountry();
>$response->getBillingCity();
>$response->getAmount();
>$response->getEmail();
>$response->getDescription();
>$response->getCurrency();
// masked credit card number
>$response->getNumber();
// Get card type
>$response->getType();
>$response->getTransactionDate();
>$response->getTransactionDate();
>$response->getSettlementDate();
// Customer reference is a reference string for a contact created in the system.
>$response->getCustomerReference();
// The recurring reference is a reference for the payment plan.
>$response->getRecurringReference();

### CODE REVIEW

**Comment by judgej** [comment]
Just to note, the failed checks are all CS issues - indentation etc. and rather a lot of them. If they could be fixed, it would be a step forwards.

**Comment by eileenmcnaughton** [comment]
Thank you - I've fixed them up now!

**Comment by eileenmcnaughton** [comment]
erm - fixed now it seems

**Comment by judgej** [comment]
New features look great, thanks.

I see a lot of duplicated code across all the new methods. Is it worth creating a new `QueryAbstractRequest` for all the shared query methods and properties? They could then perhaps all go into a `Messages\Query` sub-namespace, since the `Messages` namespace is beginning to get *very* crowded - it's something I did for [Wircard](https://github.com/academe/Omnipay-Wirecard/tree/master/src/Message).

**Comment by eileenmcnaughton** [comment]
I'm happy to do that - are there any other thoughts / changes you have before I tackle it?

**Comment by eileenmcnaughton** [comment]
I've pushed a re-org on this

**Comment by judgej** [comment]
The failed tests are all code-sniffer things - a few spaces and newlines need to be added to half a dozen files to fix those.

**Comment by eileenmcnaughton** [comment]
I got it past the cs - one hurdle down

**Comment by eileenmcnaughton** [comment]
Ok - CI is passing now

File: src/Message/Query/AIMAbstractQueryRequest.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+namespace Omnipay\AuthorizeNet\Message\Query;
+
+/**
+ * Authorize.Net AIM Abstract Request
+ */
+abstract class AIMAbstractQueryRequest extends AIMAbstractRequest
+{
+    protected $limit = 1000;
+    protected $offset = 1;
```
[the remark this answers is not in the document]
Comment (judgej) [reply src/Message/Query/AIMAbstractQueryRequest.php]: [Looking at the docs](https://developer.authorize.net/api/reference/#recurring-billing-get-a-list-of-subscriptions), Authorize.Net call this field "offset" and then state it is a "page number". So an "offset" of 1 is page 1 and so has an actual offset of zero. That's crazy design, but that's where we are!

And "limit" is, of course, "page size". So the first record from page 2 will change depending on the limit; the limit is not *just* a limit - it defines how many records are skipped before the first record is returned. It's clear, and would be intuitive if a different language was used.

**Comment by judgej** [comment]
PR now merged, thank you for this effort. Please confirm when the master branch is working as expected on your system, and I'll tag it. It'll probably be a 2.6.0 release as it adds a lot of new functionality.

**Comment by eileenmcnaughton** [comment]
Thanks for merging this! We have been using just these functions but haven't managed to migrate to Omnipay for the Authorize.net payment processing due to https://github.com/thephpleague/omnipay-authorizenet/issues/50. I think I'm going to create my own AuthorizeNet2 class that extends this & just overrides the function rather than deploying a fork of this.

**Comment by eileenmcnaughton** [comment]
(note that I'm pretty sure it's possible to use a query function to get the additional details in real time to process a refund - so the reasons that the json format rather than the merchant unique id for the transaction, is returned for getTransactionReference can be otherwise resolved)

**Comment by dmgctrlr** [comment]
Hi @eileenmcnaughton 
I've been trying to use your `$gateway->query(array('startTimestamp' => strtotime('yesterday'), 'endTimeStamp' => 'strtotime('now')));` example. I'm receiving a InvalidRequestException. I believe this is due to the AIMAbstractRequest class adding a transaction type. Do we perhaps need to add a method override for addTransactionType in the QueryRequest class?

In addition, it appears that in the AIMGateway class, calls to queryBatch, queryBatchDetail and queryDetail are not pointing to the correct files.

I may be misusing this, so please forgive my provisional understanding!

Sorry if this is not the correct place to post this, feel free to move it if required.

**Comment by judgej** [comment]
Thanks @dmgctrlr - I'll open a new issue to tackle these. If you have any further details of what you were trying to do, then please add it to the new issue. I'll fix the invalid namespaces in the meantime.

**Pull request merged**